### PR TITLE
Ddce 763 Charity’s authorised officials

### PR DESCRIPTION
**Pull request by prameya-phatak**
# DDCE-(763)

##New feature

Charity’s authorised officials

## PR Suggestions
- Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?
- Have you done a visual check of the changes?
- Is there any still commented or unused code? Lingering printlns?
- Have things been implemented in a complicated way?
- Are the test still over the coverage threshold?
- Does the compiler throw a lot of warnings? 
- Have methods and tests been named clearly?
- Were there any changes in the config? Do changes need to be made in app-config-???

## Checklist PR Raiser
 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [x]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

## Checklist PR Reviewer
 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [x]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: app/views/components/linkAsButton.scala.html [hunk 1]
```diff
@@ -17,11 +17,11 @@
...
-    id: Option[String] = None,
+link: String,
+messageKey: String,
+id: Option[String] = None,
 )(implicit messages: Messages)
 
 <a href="@link" role="button" class="govuk-button govuk-button" @id.map(x => s"id=$x")>
     @messages(messageKey)
-</a>
+</a>
```
Comment (abhijeetardale) [inline app/views/components/linkAsButton.scala.html:27]: new line please

File: conf/messages.cy [hunk 2]
```diff
@@ -1,602 +1,611 @@
...
+error.integer = Give an answer in whole numbers
+error.non_numeric = Give a value using only numbers
+error.number = Enter a valid number
+error.required = Enter a value
+error.summary.title = There is a problem
 
 
 # Sitewide Messages
 # ----------------------------------------------------------
-site.back = (W)Back(W)
```
Comment (abhijeetardale) [inline conf/messages.cy:25]: please revert all these, you are overriding the welsh translation changes

File: test/controllers/authorisedOfficials/CharityAuthorisedOfficialsControllerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        bind[UserAnswerRepository].toInstance(mockUserAnswerRepository),
+        bind[AuthIdentifierAction].to[FakeAuthIdentifierAction]
+      )
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+    reset(mockUserAnswerRepository)
+  }
+
+  val view: CharityAuthorisedOfficialsView = injector.instanceOf[CharityAuthorisedOfficialsView]
```
Comment (abhijeetardale) [inline test/controllers/authorisedOfficials/CharityAuthorisedOfficialsControllerSpec.scala:49]: use private for local variables

File: conf/app.routes [hunk 4]
```diff
@@ -19,6 +19,7 @@ GET        /keepalive                                   controllers.IndexControl
 ->         /operations-and-funds                        operationsAndFunds.Routes
 ->         /officials-and-nominees                      officialsAndNominees.Routes
 ->         /address                                     address.Routes
+->         /officials-and-nominees                      officialsAndNominees.Routes
```
Comment (abhijeetardale) [inline conf/app.routes:22]: please check its added twice

**Comment by prameya-phatak** [comment]
Review comments are closed

**Pull request merged**